### ISSUE TITLE
Ensure cursor is not obscured by iOS onscreen keyboard

### ISSUE THREAD

**Issue by tomsci** [inseven/opolua#425]
By implementing `Program.textEditor()` which provides hints to the Swift side about where any text cursor on screen is, and maybe moving the program view up to ensure it's not obscured by the onscreen keyboard.

**Comment by jbmorley** [comment]
Yeah. There's still a bunch of little tweaks to be done to the animation and heuristics for ensuring the text box avoids the keyboard. You're right in that this is a change from the shipping code as we didn't have native text boxes and were kicking out to an iOS native dialog.

**Comment by jbmorley** [comment]
I expect the navigation bar to overlap the app view, but it looks like a weird little iOS bug that we've lost the 'thin material' background for the navigation bar. Good to get it recorded here though as I'd already forgotten about it. 🙃
